src/api/v2/transactions.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use avail_subxt::{
+	api::{self, runtime_types::sp_core::bounded::bounded_vec::BoundedVec},
+	avail,
+	primitives::AvailExtrinsicParams,
+	AvailConfig,
+};
+use sp_core::H256;
+use subxt::{ext::sp_core::sr25519::Pair, tx::PairSigner};
+
+#[async_trait]
+pub trait Submit {
+	async fn submit(&self, data: Vec<u8>) -> Result<H256>;
+}
+
+#[derive(Clone)]
+pub struct RawSubmit {
+	pub node_client: avail::Client,
+	pub app_id: u32,
+}
+
+// NOTE: Replace this with avail::PairSigner after implenting required traits in avail-subxt
+pub type AvailSigner = PairSigner<AvailConfig, Pair>;
+
+#[derive(Clone)]
+pub struct SignAndSubmit {
+	pub node_client: avail::Client,
+	pub app_id: u32,
+	pub pair_signer: AvailSigner,
+}
+
+#[async_trait]
+impl Submit for RawSubmit {
+	async fn submit(&self, data: Vec<u8>) -> Result<H256> {
+		self.node_client
+			.rpc()
+			.submit_extrinsic(data)
+			.await
+			.context("Cannot submit transaction")
+	}
+}
+
+#[async_trait]
+impl Submit for SignAndSubmit {
+	async fn submit(&self, data: Vec<u8>) -> Result<H256> {
+		let data_transfer = api::tx().data_availability().submit_data(BoundedVec(data));
+		let extrinsic_params = AvailExtrinsicParams::new_with_app_id(self.app_id.into());
+		self.node_client
+			.tx()
+			.sign_and_submit(&data_transfer, &self.pair_signer, extrinsic_params)
+			.await
+			.context("Cannot sign and submit transaction")
+	}
+}
+
+#[derive(Clone)]
+pub enum Submitter {
+	Raw(RawSubmit),
+	SignAndSubmit(Box<SignAndSubmit>),
+}
+
+#[async_trait]
+impl Submit for Submitter {
+	async fn submit(&self, data: Vec<u8>) -> Result<H256> {
+		match self {
+			Self::Raw(submitter) => submitter.submit(data).await,
+			Self::SignAndSubmit(submitter) => submitter.submit(data).await,
+		}
+	}
+}
+
+pub fn submitter(
+	node_client: avail::Client,
+	app_id: u32,
+	pair_signer: Option<AvailSigner>,
+) -> Submitter {
+	match pair_signer {
+		None => Submitter::Raw(RawSubmit {
+			node_client,
+			app_id,
+		}),
+		Some(pair_signer) => Submitter::SignAndSubmit(Box::new(SignAndSubmit {
+			node_client,
+			app_id,
+			pair_signer,
+		})),
+	}
+}
```

Comment:
Is it a constructor for `Submitter`?

Reply:
You are right, but it was removed anyway :)